### PR TITLE
Cleanup old documentation and adding backend configuration documentation.

### PR DESCRIPTION
**Pull request by ipochi**

### CODE REVIEW

File: docs/configuration-reference/backend/backend.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+## Contents
+
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Configuration](#configuration)
+* [Argument reference](#argument-reference)
+
+## Introduction
+
+Lokomotive installer supports local or remote backend [S3 only] for storing terraform state.
```
Comment (invidian) [inline docs/configuration-reference/backend/backend.md]:
```suggestion
Lokomotive installer supports local or remote backend [S3 only] for storing Terraform state.
```

File: docs/configuration-reference/backend/backend.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Backend configuration guide
```
Comment (invidian) [inline docs/configuration-reference/backend/backend.md]:
```suggestion
# Backend configuration reference
```

File: docs/configuration-reference/backend/backend.md (hunk 1 above)
Comment (invidian) [reply docs/configuration-reference/backend/backend.md]: Hm, phrase "Lokomotive installer" sounds weird to me. Should we just say `Lokomotive` or just `lokoctl` maybe?

File: docs/configuration-reference/backend/backend.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  ```hcl
+  backend "local" {
+    path = "terraform.tfstate"
+  }
+  ```
+
+### S3 backend
+
+  ```hcl
+  backend "s3" {
```
Comment (invidian) [inline docs/configuration-reference/backend/backend.md]:
```suggestion
backend "s3" {
  bucket         = "<bucket_name>"
  key            = "<path_in_s3_bucket>"
  region         = "<aws_region>"
  aws_creds_path = "<aws_credentials_file_path>"
  dynamodb_table = "<dynamodb_table_name>"
}
```

File: docs/configuration-reference/backend/backend.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+backend "local" {
+  path = "terraform.tfstate"
+}
+```
+
+### S3 backend
+
+```hcl
+backend "s3" {
+  bucket = "<bucket_name>"
```
Comment (invidian) [inline docs/configuration-reference/backend/backend.md]: This is still not formatted correctly it seems.

File: docs/configuration-reference/backend/backend.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+## Configuration
+
+To use a backend, we need to define a configuration in the `.lokocfg` file.
+
+Example configuration file:
+
+### Local backend
+
+```hcl
+backend "local" {
```
Comment (invidian) [inline docs/configuration-reference/backend/backend.md]: Actually, maybe both backends should have separate file for documentation?

Comment (ipochi) [reply docs/configuration-reference/backend/backend.md]: made changes regarding this.

Comment (ipochi) [reply docs/configuration-reference/backend/backend.md]: @invidian separated into local.md and s3.md

File: docs/configuration-reference/backend/local.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+## Contents
+
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Configuration](#configuration)
+* [Argument reference](#argument-reference)
+
+## Introduction
+
+Lokomotive supports local backend for storing Terraform state.
```
Comment (invidian) [inline docs/configuration-reference/backend/local.md]: I think we should have 3 small docs. One for backend block configuration, one for local and one for s3. then backend block configuration document should just refer to the available types and their documentation.

This way, the hierarchy is nicely represented.

Comment (ipochi) [reply docs/configuration-reference/backend/local.md]: the third document (which then refers to local and s3) would be very small only a few lines.

Having the title `local.md` and `s3.md` under backend  directory I think suffices the hierarchy.

Comment (invidian) [reply docs/configuration-reference/backend/local.md]: :+1: I won't push on that, It's good enough what we have right now.

**Comment by invidian** [comment]
Rebase needed.

**Comment by iaguis** [comment]
Rebased.

**Pull request merged**